src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -64,9 +64,9 @@ bool C2Compiler::init_c2_runtime() {
       OptoReg::vm2opto[i] = OptoReg::Bad;
   }
 
-  for( OptoReg::Name i=OptoReg::Name(0); i<OptoReg::Name(REG_COUNT); i = OptoReg::add(i,1) ) {
+  for (OptoReg::Name i=OptoReg::Name(0); i<OptoReg::Name(REG_COUNT); i = OptoReg::add(i,1)) {
     VMReg r = OptoReg::as_VMReg(i);
-    if (r->is_valid()) {
+    if (r != nullptr && r->is_valid()) {
```

Comment:
Instead of changing shared code, how about changing ppc.ad to use VMRegImpl::Bad() instead of NULL?

Reply:
That makes more sense. I've changed it. Thanks!